### PR TITLE
NodeJS: Remove CallOptions as a class, allow Objects.

### PR DESCRIPTION
**Pull request by jmuk**
Fixes https://github.com/googleapis/gax-nodejs/issues/35.

### CODE REVIEW

**Comment by jmuk** [comment]
See also: https://github.com/googleapis/gax-nodejs/pull/37

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java [hunk 1]
```diff
@@ -332,9 +346,15 @@ private String returnTypeComment(Method method, MethodConfig config) {
     for (Field field : config.getRequiredFields()) {
       paramTypesBuilder.append(fieldParamComment(field, null, false));
     }
+    paramTypesBuilder.append(
+        "@param {Object=} options\n"
+            + "  Optional parameters. You can override the default settings for this call, e.g, timeout,\n"
+            + "  retries, paginations, etc. See [gax.CallOptions]{@link "
```
Comment (geigerj) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: This link does not work for me. (It pulls up a page, but there is no `#CallOptions` section.)

File: src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline [hunk 2]
```diff
@@ -292,16 +291,18 @@ LibraryServiceApi.prototype.matchBookFromArchivedBookName =
...
-  var args = arguejs({
-    shelf: Object,
-    options: [gax.CallOptions],
-    callback: [Function]
-  }, arguments);
+LibraryServiceApi.prototype.createShelf = function createShelf(
+    shelf,
+    options,
+    callback) {
+  if (options instanceof Function && callback === undefined) {
```
Comment (geigerj) [inline src/test/java/com/google/api/codegen/testdata/nodejs_doc_main_library.baseline]: This looks strange to my eyes, and it produces a lot of repeated code. Is this more JS-idiomatic than just forcing people to pass an empty object as `options` if they want to specify `callback` but not `options`?

File: src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java [hunk 3]
```diff
@@ -332,9 +346,15 @@ private String returnTypeComment(Method method, MethodConfig config) {
     for (Field field : config.getRequiredFields()) {
       paramTypesBuilder.append(fieldParamComment(field, null, false));
     }
+    paramTypesBuilder.append(
+        "@param {Object=} options\n"
+            + "  Optional parameters. You can override the default settings for this call, e.g, timeout,\n"
```
Comment (geigerj) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: Is it possible to make it more clear that `foo` is a possible property of `options` iff `foo` is a property of `CallOptions`, as opposed to there being a property of `options` which is an object of type `CallOptions`?

Comment (jmuk) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: Added to the list of keyword -- so it can be distinguished.

Comment (jmuk) [inline src/main/java/com/google/api/codegen/nodejs/NodeJSGapicContext.java]: Added to GAX.

**Pull request merged**